release/benchmark-worker-startup/benchmark_worker_startup.py

```diff
@@ -0,0 +1,363 @@
+                        )
+
+                        test = TestConfiguration(
+                            num_jobs=num_jobs,
+                            num_runs_per_job=num_runs_per_job,
+                            num_tasks_or_actors_per_run=num_tasks_or_actors_per_run,
+                            with_tasks=with_tasks,
+                            with_gpu=with_gpu,
+                            with_runtime_env=with_runtime_env,
+                            import_to_try=import_to_try,
+                            num_cpus_in_cluster=num_cpus_in_cluster,
+                            num_gpus_in_cluster=num_gpus_in_cluster,
+                            num_nodes_in_cluster=1,
+                        )
+                        tests.add(test)
+
+    return tests
+
+
+@dataclass(eq=True, frozen=True)
+class TestConfiguration:
+    num_jobs: int
+    num_runs_per_job: int
+    num_tasks_or_actors_per_run: int
+    with_gpu: bool
+    with_tasks: bool
+    with_runtime_env: bool
+    import_to_try: str
+    num_cpus_in_cluster: int
+    num_gpus_in_cluster: int
+    num_nodes_in_cluster: int
+
+    def __repr__(self):
+        with_gpu_str = "with_gpu" if self.with_gpu else "without_gpu"
+        executable_unit = "tasks" if self.with_tasks else "actors"
+        cold_or_warm_start = "cold" if self.num_jobs > 1 else "warm"
+        with_runtime_env_str = (
+            "with_runtime_env" if self.with_runtime_env else "without_runtime_env"
+        )
+        single_node_or_multi_node = (
+            "single_node" if self.num_nodes_in_cluster == 1 else "multi_node"
+        )
+        import_torch_or_none = (
+            "import_torch" if self.import_to_try == "torch" else "no_import"
+        )
+
+        return "-".join(
+            [
+                f"seconds_to_{cold_or_warm_start}_start_{self.num_tasks_or_actors_per_run}_{executable_unit}",
+                import_torch_or_none,
+                with_gpu_str,
+                single_node_or_multi_node,
+                with_runtime_env_str,
+                f"{self.num_cpus_in_cluster}_CPU_{self.num_gpus_in_cluster}_GPU_cluster",
+            ]
+        )
+
+
+async def run_and_stream_logs(
+    metrics_actor_name, metrics_actor_namespace, test: TestConfiguration
+):
+    """
+    Run a particular test configuration by invoking ./test_single_configuration.py.
+    """
+    client = JobSubmissionClient("http://127.0.0.1:8265")
+    entrypoint = generate_entrypoint(metrics_actor_name, metrics_actor_namespace, test)
+
+    for _ in range(test.num_jobs):
+        print(f"Running {entrypoint}")
+
+        if not test.with_runtime_env:
+            # On non-workspaces, this will run as a job but without a runtime env.
+            subprocess.check_call(entrypoint, shell=True)
+        else:
+            job_id = client.submit_job(
+                entrypoint=entrypoint,
+                runtime_env={"working_dir": "./"},
```

Comment:
Sang says that Anyscale jobs don't always use this (I thought otherwise but could be wrong). We should confirm.

Reply:
We should go for larger working dir later to emulate customer workload